c/meterpreter/source/extensions/stdapi/server/net/socket/tcp_server.c

```diff
@@ -364,7 +444,7 @@ DWORD request_net_tcp_server_channel_open(Remote * remote, Packet * packet)
 			BREAK_ON_WSAERROR("[TCP-SERVER] request_net_tcp_server_channel_open. WSAEventSelect failed");
 		}
 
-		ctx->ipv6 = !v4Fallback;
+		ctx->ipv6 = ai_family == AF_INET6;

```

Comment:
You could move this info a section of the code inside your `else if (ai_family == AF_INET6)` to prevent checking this value twice and optimize things a bit more? I don't think this would affect the `bind` operation either since you are only passing the value of `ctx->fd` there?